package/src/components/Chat/hooks/useIsOnline.ts

```diff
@@ -73,23 +72,27 @@ export const useIsOnline = <
     let unsubscribeNetInfo: NetInfoSubscription;
     const setNetInfoListener = () => {
       unsubscribeNetInfo = NetInfo.addEventListener((netInfoState) => {
-        if (netInfoState === false && !client.wsConnection?.isHealthy) {
+        if (!netInfoState && !client.wsConnection?.isHealthy) {
           setConnectionRecovering(true);
           setIsOnline(false);
         }
-        notifyChatClient(netInfoState);
+        const { isConnected, isInternetReachable } = netInfoState;
+        notifyChatClient(
+          isInternetReachable !== null ? isInternetReachable && isConnected : isConnected,
+        );
       });
     };
 
     const setInitialOnlineState = async () => {
-      const status = await NetInfo.fetch();
-      if (isMounted.current) {
-        setIsOnline(status);
-        notifyChatClient(status);
-      }
+      const { isConnected } = await NetInfo.fetch();
+      setIsOnline(isConnected);
+      notifyChatClient(isConnected);
     };
 
-    setInitialOnlineState();
+    if (isMounted.current) {
```

Comment:
```
    if (isMounted.current) {
```

I would keep inside the async function, that's better because when we set the state it is guaranteed that its mounted. 

Now it is not.. after mount we fetch then wait then set the state.. this change is unnecessary

Reply:
If we do it outside, we will not have to `fetch()` API unnecessarily; that is why I have made this change. The function `setInitialOnlineState` will only be called if its mounted and this way state is also set when its mounted.